backend/src/services/worker/worker.service.ts

```diff
@@ -196,6 +226,31 @@ class Worker {
       throw new EAMError();
     }
 
+    const master = await this.#masterService.getMasterById(id);
+
+    if (master) {
+      throw new EAMError({
+        status: HttpCode.DENIED,
+        message: ExceptionMessage.MASTER_DELETE,
+      });
+    }
+
+    const instances = await this.#instanceRepository.getByWorkerId(id);
+    const space = await this.#spaceRepository.getByWorkerId(id);
+    const slcFunctions = await this.#slcFunctionRepository.getByWorkerId(id);
+
+    for (const id of instances) {
+      await this.#instanceService.delete(id);
+    }
```

Comment:
```suggestion
   await Promise.all(instances.map(instance => this.#instanceService.delete(instance .id));
```

Reply:
`for await` processes requests sequentially. `Promise all` processes concurrently. 
You can find more about differences [here](https://www.linkedin.com/pulse/await-vs-promiseall-mohamed-kamel/)

Reply:
agreed. prefer promise.all for all